syntax = "proto3";

package onode;

message ControlMessage {
  enum MessageType {
    REGISTER = 0;
    REGISTER_RESPONSE = 1;
    PING = 2;
    PONG = 3;
    UPDATE_NEIGHBORS = 4;
  }

  MessageType type = 1;
  string node_id = 2;
  repeated NeighborInfo neighbors = 3;
  int32 control_port = 4;
  int32 data_port = 5;
}

message NeighborInfo {
  string node_id = 1;
  string ip = 2;
  int32 control_port = 3;
  int32 data_port = 4;
}

message ContentRequest {
  string content_id = 1;     
  string client_id = 2;   
}

message ContentResponse {
  string content_id = 1;    
  string source_ip = 2;     
  int32 source_port = 3;  
}
